### ISSUE TITLE
Introduce `@RangeSource` for parameterized tests

### ISSUE THREAD

**Issue by ghost** [junit-team/junit5#851]
## Overview

I was reading up on `@ParameterizedTest` and considering the `@ValueSource` test.

If you wanted to run a test over a large range of values, it could be quite a lengthy process to write out each test value.

Would it be worth including a `@RangeSource` test? Something like:

```java
@ParameterizedTest
@RangeSource(from = 100, to = 1000, skip = 10)
void testWithRangeSource(int argument) {
	assertNotNull(argument);
}
```

I'm happy to have a go at creating a PR for this if it would be useful.

**Comment by marcphilipp** [comment]
I think this is already sufficiently covered by using a method source:

```java
@ParameterizedTest
@MethodSource("range")
void testWithRangeMethodSource(int argument) {
	assertNotEquals(-1, argument);
}

static Stream<Integer> range() {
	return IntStream.range(0, 1000).skip(10).boxed();
}
```

However, we could add support for `IntStream` etc. directly.

**Comment by marcphilipp** [comment]
@smoyer64 @nicolaiparlog Maybe something for [junit-pioneer](https://github.com/junit-pioneer/junit-pioneer)?

**Comment by marcphilipp** [comment]
I've added support for returning a "primitive" stream directly in 66cfa8aecb6dcadacd4c2c09eaa3f1f3855db080.

**Comment by marcphilipp** [comment]
**Team Decision:** Closed in favor of junit-pioneer/junit-pioneer#44.
